app/Http/Controllers/SensorController.php

```diff
@@ -0,0 +1,123 @@
+<?php 
+
+namespace App\Http\Controllers;
+
+use App\Sensor;
+use App\DataTables\SensorDataTable;
+use Illuminate\Http\Request;
+use Validator;
+
+class SensorController extends Controller 
+{
+    /**
+    * Create a new controller instance.
+    *
+    * @return void
+    */
+    public function __construct()
+    {
+        $this->middleware('auth');
+    }
+
+    /**
+    * Display a listing of the resource.
+    *
+    * @param  SensorDataTable   $dataTable
+    * @return Response
+    */
+    public function index(SensorDataTable $dataTable)
+    {
+        return $dataTable->render('sensor.index');
+    }
+
+    /**
+    * Show the form for creating a new resource.
+    *
+    * @return Response
+    */
+    public function create()
+    {
+        return view('sensor.create');
+    }
+
+    /**
+    * Store a newly created resource in storage.
+    *
+    * @param Request $request
+    * @return Response
+    */
+    public function store(Request $request)
+    {
+        request()->validate([
+            'device_id' => 'required|integer|exists:devices,id',
+            'name' => 'required|string|max:190',
+            'type' => 'required|string|max:190'
+        ]);
+
+        $query = Sensor::create($request->all());
+
```

Comment:
You need to create a starting sensordata value for the newly created sensor. If you don't the sensor datatable returns an error because the value is null.

Reply:
The device dictates sensors. Instead of making dummy sensors we use the null coalescence operator when accessing relationships that may not exist. e.g. ``` ($sensordata->sensor->device->name ?? 'null') ``` replaces a missing device name with null.